themes-default/slim/views/vue-components/app-link.mako

```diff
@@ -30,6 +31,17 @@ Vue.component('app-link', {
         config() {
             return this.$store.state.config;
         },
+        indexerName() {
+            const { indexerId, config } = this;
+            if (!indexerId) {
+                return '';
+            }
+            return Object.keys(config.indexers).filter(indexer => {
+                if (config.indexers[indexer].id === parseInt(indexerId, 10)) {
+                    return config.indexers[indexer].name;
+                }
+            })[0];
```

Comment:
1. The keys of `config.indexers` don't contain the indexer names, it's `config.indexers.config.indexers`. see below.
    It's because of the MEDUSA.config structure. When we stop using it we should be able to fix that...
2. This can use `find` instead of filter, and be much simpler.
    This array iteration is so confusing. We don't return the name, we return the key (which is the identifier).
    `name=TVDBv2 | identifier=tvdb`
```js
            const { indexers } = config.indexers.config; // This is weird, I know...
            return Object.keys(indexers).find(indexer => indexers[indexer].id === parseInt(indexerId, 10));

            const { indexers } = config.indexers.config; // This is weird, I know...
            const matched = Object.keys(indexers).find(indexer => indexer.id === parseInt(indexerId, 10));
            return matched ? matched.name : undefined;
```

-----
[1]
```json
{ store.state
  "config": {
    "indexers": {
      "config": {
        "indexers": {
          "tvdb": {
            "xemOrigin": "tvdb",
            "apiParams": {
              "useZip": true,
              "language": "en"
            },
            "showUrl": "https://www.thetvdb.com/dereferrer/series/",
            "id": 1,
            "icon": "thetvdb16.png",
            "name": "TVDBv2",
            "scene_loc": "https://cdn.pymedusa.com/scene_exceptions/scene_exceptions_tvdb.json",
            "enabled": true,
            "baseUrl": "https://api.thetvdb.com/",
            "mappedTo": "tvdb_id",
            "identifier": "tvdb"
          },
          "tmdb": {
            "xemOrigin": null,
            "apiParams": {
              "useZip": true,
              "language": "en"
            },
            "showUrl": "https://www.themoviedb.org/tv/",
            "id": 4,
            "icon": "tmdb16.png",
            "name": "TMDB",
            "scene_loc": "https://cdn.pymedusa.com/scene_exceptions/scene_exceptions_tmdb.json",
            "enabled": true,
            "baseUrl": "https://www.themoviedb.org/",
            "mappedTo": "tmdb_id",
            "identifier": "tmdb"
          },
          "tvmaze": {
            "xemOrigin": null,
            "apiParams": {
              "useZip": true,
              "language": "en"
            },
            "showUrl": "http://www.tvmaze.com/shows/",
            "id": 3,
            "icon": "tvmaze16.png",
            "name": "TVmaze",
            "scene_loc": "https://cdn.pymedusa.com/scene_exceptions/scene_exceptions_tvmaze.json",
            "enabled": true,
            "baseUrl": "http://api.tvmaze.com/",
            "mappedTo": "tvmaze_id",
            "identifier": "tvmaze"
          }
        },
        "main": {
          "externalMappings": {<truncated>},
          "statusMap": {<truncated>},
          "traktIndexers": {<truncated>},
          "validLanguages": [<truncated>],
          "langabbvToId": {<truncated>}
        }
      }
    }
  }
}
```